src/coord/cartesian/Grid.ts

```diff
@@ -85,18 +86,68 @@ class Grid implements CoordinateSystemMaster {
         return this._rect;
     }
 
+    calAxisNiceSplitNumber(axes: Axis2D[]) {
+        const splitNumber: number[] = [];
+        const niceAxisExtents: number[] = [];
+        each(axes, axis => {
+            niceScaleExtent(axis.scale, axis.model);
+            if (axis.type !== 'value') {
+                return;
+            }
+            const extent = axis.scale.getExtent();
+            const interval = (axis.scale as IntervalScale).getInterval();
+            splitNumber.push(
+                Math.floor((extent[1] - extent[0]) / interval)
+            );
+            const axisExtent = axis.getExtent();
+            niceAxisExtents.push((axisExtent[1] - axisExtent[0]) * interval / extent[1]);
+        });
+
+        return splitNumber;
+    }
+
+    resetAxisExtent(axes: Axis2D[], splitNumber: number[], maxSplitNumber: number): number[] {
+        const finalSplitNumber: number[] = [];
+        each(axes, function (axis, index) {
+            if (axis.type !== 'value') {
+                return;
+            }
+
+            if (!(maxSplitNumber - splitNumber[index])) {
+                finalSplitNumber.push(maxSplitNumber);
+                return;
+            };
+
+            const extent = axis.scale.getExtent();
+            const interval = (axis.scale as IntervalScale).getInterval();
+            if (((extent[1] - extent[0]) % interval) === 0) {
+                extent[1] = extent[1] + (maxSplitNumber - splitNumber[index]) * interval;
+                niceScaleExtent(axis.scale, axis.model, extent, maxSplitNumber);
+            }
+            else {
+                (axis.scale as IntervalScale).niceTicks(
+                    maxSplitNumber + 1, null, null,
+                    ((extent[1] - extent[0]) - ((extent[1] - extent[0]) % maxSplitNumber)) / maxSplitNumber
+                );
```

Comment:
`extent[1] - extent[0]` should be rounded here to avoid rounding error.
```js
numberUtil.round(extent[1] - extent[0]);
```

The result interval should also be rounded.